build.nix

```diff
@@ -239,6 +239,7 @@ let
       # threads as there are cores. This is often too much parallelism so we
       # reduce it to $NIX_BUILD_CORES if not specified by the caller.
       export RUST_TEST_THREADS="''${RUST_TEST_THREADS:-$NIX_BUILD_CORES}"
+      export CARGO_TARGET_DIR="$out"
```

Comment:
I think you'll want `="out"` as opposed to `$out`! we want the directory `out`, not the build's output dir. Binaries are _then_ copied from `out` to `$out/bin`.

Reply:
Done.